client/settings/payment-settings/general-settings-section.js

```diff
@@ -21,76 +20,48 @@ const GeneralSettingsSection = () => {
 	const [ enableStripe, setEnableStripe ] = useState( false );
 	const [ enableTestMode, setEnableTestMode ] = useState( false );
 
-	const toggleEnableStripe = () => setEnableStripe( ! enableStripe );
-	const toggleEnableTestMode = () => setEnableTestMode( ! enableTestMode );
-
-	const CheckboxDescription = styled.span`
```

Comment:
We shouldn't declare styled components within the body of another component. This can cause a memory leak, as the component gets re-rendered.
Styled components should always be part of the global scope.

See: https://styled-components.com/docs/faqs#why-should-i-avoid-declaring-styled-components-in-the-render-method

Reply:
I think this explains why my test failed [before](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1800#discussion_r704847029).

The test in here https://github.com/woocommerce/woocommerce-gateway-stripe/blob/efc7afd432ce6f0e9faeefc7f6730d9b1ebd6561/client/settings/payment-settings/__tests__/general-settings-section.test.js#L17-L28 can probably use the variable now, since it no longer discards the entire component 👍 
```
		const enableStripe = screen.getByLabelText( 'Enable Stripe' );
		const enableTestMode = screen.getByLabelText( 'Enable test mode' );

		expect( enableStripe ).not.toBeChecked();
		expect( enableTestMode ).not.toBeChecked();

		userEvent.click( enableStripe );

		expect( enableStripe ).toBeChecked();
		expect( enableTestMode ).not.toBeChecked();

		userEvent.click( enableTestMode );

		expect( enableStripe ).toBeChecked();
		expect( enableTestMode ).toBeChecked();
```

Reply:
Ah sure - I don't mind it either way, but here's the change: 251cf396

Reply:
I didn't mean that it needed a change, but more so on why my `const `didn't work before. 👍

Reply:
Ah, gotcha 😆 sorry for the misunderstanding!

Reply:
😆  no need to apologize! i didn't need to make that extra commit haha. Thanks for considering it!